Foundation/URLSession/URLSessionConfiguration.swift

```diff
@@ -119,7 +119,15 @@ open class URLSessionConfiguration : NSObject, NSCopying {
     open class var `default`: URLSessionConfiguration {
         return URLSessionConfiguration()
     }
-    open class var ephemeral: URLSessionConfiguration { NSUnimplemented() }
+
+    open class var ephemeral: URLSessionConfiguration {
+        // Return a new ephemeral URLSessionConfiguration every time this property is invoked
+        // TODO: urlCache and urlCredentialStorage should also be ephemeral/in-memory
+        // URLCache and URLCredentialStorage are still unimplemented
+        let ephemeralConfiguration = URLSessionConfiguration()
+        ephemeralConfiguration.httpCookieStorage = HTTPCookieStorage.ephemeralStorage()
```

Comment:
I just printed all the public property of ephemeral configuration, and noticed some minor discrepancies of values between the existing implementation and yours,

- value for `httpMaximumConnectionsPerHost` is 4, but it is 6 in yours
- `protocolClass` array contains `[_NSURLHTTPProtocol, _NSURLDataProtocol, _NSURLFTPProtocol, _NSURLFileProtocol, NSAboutURLProtocol]`, but it is just `[SwiftFoundation._HTTPURLProtocol]` in yours.

Hope this helps.

Reply:
That's right. `protocolClass` array contains all the default `native` protocol implementations that come with `URLSession`. These are `http`, `ftp`, `file` and `data`. We currently have only `http` implemented and there is a pull request for `ftp` too. This array will be populated as and when the respective implementations are available. Thanks for noticing this, though!

Reply:
@karthikkeyan I checked the value for `httpMaximumConnectionsPerHost` on Darwin with the latest dev snapshot. It is 6.
```
Welcome to Apple Swift version 5.0-dev (LLVM 082dec2e22, Swift 1e5637f841).
Type :help for assistance.
  1> import Foundation
  2> URLSessionConfiguration.ephemeral.httpMaximumConnectionsPerHost
$R0: Int = 6
```